FluentStorage.SFTP/SshNetSftpBlobStorage.cs

```diff
@@ -263,19 +265,27 @@ public SshNetSftpBlobStorage(string host, string username, params PrivateKeyFile
 
 			SftpClient client = GetClient();
 
-			IEnumerable<SftpFile> directoryContents = await client.ListDirectoryAsync(options.FolderPath);
+			var folder = StoragePath.Combine(RootDirectory, StoragePath.Normalize(options.FolderPath));
+			IEnumerable<SftpFile> directoryContents = await client.ListDirectoryAsync(folder);
 
-			IEnumerable<Blob> blobCollection = directoryContents
+			List<Blob> blobCollection = directoryContents
 			   .Where(dc => (options.FilePrefix == null || dc.Name.StartsWith(options.FilePrefix))
 							&& (dc.IsDirectory || dc.IsRegularFile || dc.OwnerCanRead)
 							&& !cancellationToken.IsCancellationRequested
 							&& dc.Name != "."
 							&& dc.Name != "..")
 			   .Take(options.MaxResults.Value)
 			   .Select(ConvertSftpFileToBlob)
-			   .Where(options.BrowseFilter);
+			   .Where(options.BrowseFilter)
+			   .ToList();
 
-			return blobCollection.ToList();
+			if (RootDirectory != null) {
```

Comment:
add a comment about what this does

Reply:
If RootDirectory is not null, then from the FullPath of each file in ListAsync, we should remove the common RootDirectory. This makes it compatible with deleting, downloading etc, as we always want a relative path. This is exactly how the Local Storage works also: all files are relative to the path=XXX folder.